src/main/java/me/zeroeightsix/kami/module/modules/misc/AntiAFK.kt

```diff
@@ -7,80 +7,138 @@ import me.zero.alpine.listener.EventHook
     private val squareWalk = register(Settings.b("SquareWalk", true))
     private val radius = register(Settings.integerBuilder("Radius").withMinimum(1).withValue(64).build())
-    private val turn = register(Settings.b("Turn", true))
+    private val inputTimeout = register(Settings.integerBuilder("InputTimeout(m)").withValue(0).withRange(0, 15).build())
 
-    private val random = Random()
-    private val squareStartCoords = intArrayOf(0, 0)
+    private var startPos = BlockPos(114514, -696969, 404)
+    private var nextActionTick = 0
     private var squareStep = 0
     private var baritoneDisconnectOnArrival = false
+    private val inputTimer = TimerUtils.TickTimer(TimerUtils.TimeUnit.MINUTES)
 
-    public override fun onEnable() {
-        if (mc.player == null) return
+    @EventHandler
+    private val receiveListener = Listener(EventHook { event: PacketEvent.Receive ->
+        if (!autoReply.value || event.packet !is SPacketChat) return@EventHook
+        val text = event.packet.chatComponent.unformattedText
+        if (text.contains("whispers: ") && !text.contains(mc.player.name)) {
+            sendServerMessage("/r I am currently AFK and using KAMI Blue!")
+        }
+    })
+
+    @EventHandler
+    private val mouseInputListener = Listener(EventHook<InputEvent.MouseInputEvent> { event: InputEvent.MouseInputEvent? ->
+        if (inputTimeout.value != 0 && isInputting()) {
+            inputTimer.reset()
+        }
+    })
 
-        baritoneDisconnectOnArrival = BaritoneAPI.getSettings().disconnectOnArrival.value
-        BaritoneAPI.getSettings().disconnectOnArrival.value = false
-        squareStartCoords[0] = mc.player.posX.toInt()
-        squareStartCoords[1] = mc.player.posZ.toInt()
+    @EventHandler
+    private val keyInputListener = Listener(EventHook<KeyInputEvent> { event: KeyInputEvent? ->
+        if (inputTimeout.value != 0 && isInputting()) {
+            inputTimer.reset()
+        }
+    })
+
+    private fun isInputting(): Boolean {
+        return (mc.gameSettings.keyBindAttack.isKeyDown
+                || mc.gameSettings.keyBindUseItem.isKeyDown
+                || mc.gameSettings.keyBindJump.isKeyDown
+                || mc.gameSettings.keyBindSneak.isKeyDown
+                || mc.gameSettings.keyBindForward.isKeyDown
+                || mc.gameSettings.keyBindBack.isKeyDown
+                || mc.gameSettings.keyBindLeft.isKeyDown
+                || mc.gameSettings.keyBindRight.isKeyDown)
     }
 
-    public override fun onDisable() {
-        if (mc.player == null) return
+    override fun getHudInfo(): String? {
+        return ((System.currentTimeMillis() - inputTimer.time) / 1000L).toString()
+    }
+
+    override fun onEnable() {
+        with(BaritoneAPI.getSettings().disconnectOnArrival) {
+            baritoneDisconnectOnArrival = value
+            value = false
+        }
+    }
 
+    override fun onDisable() {
+        startPos = BlockPos(114514, -696969, 404)
```

Comment:
Why are we doing this?

Reply:
Just a dummy value telling that the blockpos is not initialized lol